### DISCUSSION TITLE
example for using timer.TOff

### DISCUSSION THREAD

**Discussion by jo47011** [mhthies/smarthomeconnect#86] [Q&A] [answered]
Can you please provide a short example on how to use the `toff` timer.  I would like a bool KNX Variable `light_table` to automatically turn off after 5 seconds.  I tried this but that does not work:

`light_table.connect(shc.timer.TOff(light_table, datetime.timedelta(seconds=5)))`

Thanks a lot...

**Comment by mhthies** [comment, marked as answer]
Hi,

The `TOff` timer is only a delay for changes to off. This means if the value of the inserted *subscribable* object changes to `False`, the `TOff`'s value will change to `False` 5 seconds later, but it will change to `True` immediately when the value of the inserted object changes to `True`. This functionality is simliar to the TOFF block in SPS programming languages (e.g. https://infosys.beckhoff.com/english.php?content=../content/1033/tcplclib_tc2_standard/74405003.html&id=).

For your purpose, the `TOff` timer is probably not the right tool. Instead, I would use a logic handler function like this:

```python
@light_table.trigger
@shc.handler(reset_origin=True)
async def switch_off_light_table(value):
    """ Switch off table light after 5s """
    if not value:
        return
    await asyncio.sleep(5)
    await light_table.write(False)
```

**Reply by jo47011** [reply]
Hi Michael,

thank you.  I knew the rule approach but was trying to use an existing approach if any.  Now I understand the Toff timer.  Thanks a lot for the explanation.  Once again thanks a lot for all your efforts.  It's really a great project.  100% pure pyhton 👍😎 I'm currently migrating from openhab and I'm really excited to get everything up and running.  So far really happy with shc.  Let me know whether you have a "buy me a coffee" link or alike.  Don't know whether I'll have enough in-depth knowledge to contribute in the near future but I'd be happy to appreciate all you work somehow.
